pypeit/core/flux_calib.py

```diff
@@ -616,73 +732,67 @@ def sensfunc_eval(wave, counts, counts_ivar, counts_mask, exptime, airmass, std_
             If you have significant telluric absorption you should be using telluric.sensnfunc_telluric. default = 0.9
 
     Returns:
-        tuple: Returns:
-
-            - sensfunc (ndarray) -- Sensitivity function with same shape as wave (nspec,)
-            - mask_sens (ndarray, bool) -- Good pixel mask for sensitivity function with same shape as wave (nspec,)
+            zeropoint (`numpy.ndarray`_): Sensitivity function with same shape as wave (nspec,)
+            mask_sens (`numpy.ndarray`_): Good pixel mask for sensitivity function with same shape as wave (nspec,)
 
     """
     # Create copy of the arrays to avoid modification and convert to
-    # electrons / s
-    wave_star = wave.copy()
-    flux_star = counts.copy() / exptime
-    ivar_star = counts_ivar.copy() * exptime ** 2
+    # Nlam = electrons/s/Angstrom
+    delta_wave = wvutils.get_delta_wave(wave, (wave > 1.0))
+    Nlam_star = counts/exptime/delta_wave
+    Nlam_ivar_star = delta_wave**2*counts_ivar*exptime**2
 
     # Extinction correction
     msgs.info("Applying extinction correction")
     extinct = load_extinction_data(longitude,latitude)
     ext_corr = extinction_correction(wave * units.AA, airmass, extinct)
     # Correct for extinction
-    flux_star = flux_star * ext_corr
-    ivar_star = ivar_star / ext_corr ** 2
-    mask_star = counts_mask
+    Nlam_star = Nlam_star * ext_corr
+    Nlam_ivar_star = Nlam_ivar_star / ext_corr ** 2
+    gpm_star = counts_mask
 
     # Interpolate the standard star onto the current set of observed wavelengths
     flux_true = interpolate.interp1d(std_dict['wave'], std_dict['flux'], bounds_error=False,
-                                     fill_value='extrapolate')(wave_star)
+                                     fill_value='extrapolate')(wave)
     # Do we need to extrapolate? TODO Replace with a model or a grey body?
     if np.min(flux_true) <= 0.:
         msgs.warn('Your spectrum extends beyond calibrated standard star, extrapolating the spectra with polynomial.')
         mask_model = flux_true <= 0
         pypeitFit = fitting.robust_fit(std_dict['wave'].value, std_dict['flux'].value,8,function='polynomial',
                                                     maxiter=50, lower=3.0, upper=3.0, maxrej=3,
                                                     grow=0, sticky=True, use_mad=True)
-        star_poly = pypeitFit.eval(wave_star.value)
+        star_poly = pypeitFit.eval(wave.value)
```

Comment:
This conditional branch crashed for me because wave is of type np.ndarray, which has no attribute `value`. I think changing this line to `star_poly = pypeitFit.eval(wave)` would fix this error.

Reply:
Good catch @finagle29. This bug is now fixed. Note that this extrapolation bit is a hack that rarely gets triggered which is why it didn't crash for other tests. If this is being triggered for your data, you are likely extrapolating the standard star in a big way and the fluxing is not reliable. You can run with the debug option and it will make a plot.